samples/linkedin/index.ts

```diff
@@ -0,0 +1,121 @@
+import { TransformApp } from "@scramjet/types";
+// import { DataStream } from "scramjet";
+import { PassThrough } from "stream";
+import csv from 'csv-parser';
+import axios, { AxiosRequestConfig } from 'axios';
+import * as cheerio from 'cheerio';
+import { XRapidAPIKey } from './config.json';
+
+const sleep = (timeMs: number) => new Promise(res => setTimeout(res, timeMs));
+const RATE_LIMIT_MS = 200;
+
+interface Company {
+    LinkedIn: string
+}
+
+/**
+ * Get company name from LinkedIn URL
+ * 
+ * @param {Company} company
+ * @returns {String} Company Name
+ */
+function getCompanyName(company: Company): String | null {
+    if (company.LinkedIn) {
+        const { pathname } = new URL(company.LinkedIn);
+
+        if (pathname.indexOf('/company/') !== 0) return null;
+
+        return pathname.match(/^\/company\/([a-zA-Z0-9\-]+)/)![1];
+    }
+
+    return null;
+}
+
+
+/**
+ * Get company data from Rapid API
+ * 
+ * @param {String} companyName 
+ * @returns {Object}
+ */
+async function getDataFromRapiAPI(companyName: String | null) {
+    if (companyName === null) return null;
+
+    const options = {
+        method: 'GET',
+        url: 'https://linkedin-companies-data.p.rapidapi.com/',
+        params: { vanity_name: companyName },
+        headers: {
+            'X-RapidAPI-Key': XRapidAPIKey,
+            'X-RapidAPI-Host': 'linkedin-companies-data.p.rapidapi.com',
+        },
+    };
+
+    return (await axios.request(options as AxiosRequestConfig)).data
+}
+
+/** 
+ * Get title, description, keywords and first 1000 characters from website
+ * 
+ * @param {String} url
+ * @returns {Object}
+ */
+async function scrape(url: string) {
+    if (!url) return null;
+
+    let html;
+    try {
+        html = (await axios.get(url)).data;
+    } catch (error) {
+        // In case of an error handle it gracefully
+        console.log(error);
+        return null;
+    }
+
+    const $ = cheerio.load(html);
+    const title = $('head > title').text();
+    const description = $('meta[name="description"]').attr('content');
+    const keywords = $('meta[name="keywords"]').attr('content');
+    const body = $('body').text().substr(0, 1000);
+
+    return { title, description, keywords, body };
+}
+
+const app: TransformApp = (
+    input
+) => {
+    const out = new PassThrough({ encoding: "utf-8" });
```

Comment:
```suggestion
    const XRapidAPIKey = this.config.key ? `${this.config.key}` : "";
    
    if (!XRapidAPIKey) throw new Error("No API key provided in config");
    
    const out = new PassThrough({ encoding: "utf-8" });
```

Reply:
I'm getting `Object is possibly 'undefined'.ts(2532)` for `this.config.key`

Additionally, shouldn't this be `appConfig` instead of `config`?

Reply:
Anyway... looks like I'm missing something (config configuration?) and I can't get this to work.

Reply:
nvm... my stupid mistake! corrected and works fine.